chef/cookbooks/provisioner/recipes/update_nodes.rb

```diff
@@ -20,6 +20,10 @@ def find_node_boot_mac_addresses(node, admin_data_net)
   result = []
   admin_interfaces = admin_data_net.interface_list
   admin_interfaces.each do |interface|
+    if interface.nil?
+      raise "Failed to find admin node interface for node #{node.fqdn}! " \
+            "Check your network barclamp proposal."
+    end
```

Comment:
Thinking more about this: no objection about this part, as it's no regression, but I think it's not enough. We do want to avoid a crash in chef-client here, as having a rogue node could result, thanks to this, in a DoS on part of the deployment infrastructure. I'll submit a patch.